File: nativelogger/src/main/java/cn/jesse/nativelogger/logger/base/ILogger.java
package cn.jesse.nativelogger.logger.base;

import cn.jesse.nativelogger.logger.LoggerLevel;

/**
 * Created by jesse on 9/6/16.
 */
public interface ILogger {
    void setTag(String tag);
    String tag();

    void setLevel(LoggerLevel level);

    boolean isDebugEnabled();

    void debug(String msg);

    void debug(String subTag, String msg);

    void debug(String subTag, String format, Object arg);

    void debug(String subTag, String format, Object argA, Object argB);

    void debug(String subTag, String format, Object... arguments);

    void debug(String subTag, Throwable t);

    boolean isInfoEnabled();

    void info(String msg);

    void info(String subTag, String msg);

    void info(String subTag, String format, Object arg);

    void info(String subTag, String format, Object argA, Object argB);

    void info(String subTag, String format, Object... arguments);

    void info(String subTag, Throwable t);

    boolean isWarnEnabled();

    void warn(String msg);

    void warn(String subTag, String msg);

    void warn(String subTag, String format, Object arg);

    void warn(String subTag, String format, Object... arguments);

    void warn(String subTag, String format, Object argA, Object argB);

    void warn(String subTag, Throwable t);

    boolean isErrorEnabled();

    void error(String msg);

    void error(String subTag, String msg);

    void error(String subTag, String format, Object arg);

    void error(String subTag, String format, Object argA, Object argB);

    void error(String subTag, String format, Object... arguments);

    void error(String subTag, Throwable t);

    boolean isEnabled(LoggerLevel level);

    void log(LoggerLevel level, String msg);

    void log(LoggerLevel level, String subTag, String msg);

    void log(LoggerLevel level, String subTag, String format, Object arg);

    void log(LoggerLevel level, String subTag, String format, Object argA, Object argB);

    void log(LoggerLevel level, String subTag, String format, Object... arguments);

    void log(LoggerLevel level, String subTag, Throwable t);

    void json(LoggerLevel level, String msg);

    void json(LoggerLevel level, String subTag, String msg);
}

File: nativelogger/src/main/java/cn/jesse/nativelogger/logger/base/ILogger.kt
package cn.jesse.nativelogger.logger.base

import cn.jesse.nativelogger.logger.LoggerLevel

/**
 * Logger 接口
 *
 * @author Jesse
 */
interface ILogger {

    fun setTag(tag: String)
    fun tag(): String

    fun setLevel(level: LoggerLevel)
    fun isDebugEnabled(): Boolean

    fun debug(msg: String)

    fun debug(subTag: String, msg: String)

    fun debug(subTag: String, format: String, arg: Any)

    fun debug(subTag: String, format: String, argA: Any, argB: Any)

    fun debug(subTag: String, format: String, vararg arguments: Any)

    fun debug(subTag: String, t: Throwable)

    fun isInfoEnabled(): Boolean

    fun info(msg: String)

    fun info(subTag: String, msg: String)

    fun info(subTag: String, format: String, arg: Any)

    fun info(subTag: String, format: String, argA: Any, argB: Any)

    fun info(subTag: String, format: String, vararg arguments: Any)

    fun info(subTag: String, t: Throwable)

    fun isWarnEnabled(): Boolean

    fun warn(msg: String)

    fun warn(subTag: String, msg: String)

    fun warn(subTag: String, format: String, arg: Any)

    fun warn(subTag: String, format: String, vararg arguments: Any)

    fun warn(subTag: String, format: String, argA: Any, argB: Any)

    fun warn(subTag: String, t: Throwable)

    fun isErrorEnabled(): Boolean

    fun error(msg: String)

    fun error(subTag: String, msg: String)

    fun error(subTag: String, format: String, arg: Any)

    fun error(subTag: String, format: String, argA: Any, argB: Any)

    fun error(subTag: String, format: String, vararg arguments: Any)

    fun error(subTag: String, t: Throwable)

    fun isEnabled(level: LoggerLevel): Boolean

    fun log(level: LoggerLevel, msg: String)

    fun log(level: LoggerLevel, subTag: String, msg: String)

    fun log(level: LoggerLevel, subTag: String, format: String, arg: Any)

    fun log(level: LoggerLevel, subTag: String, format: String, argA: Any, argB: Any)

    fun log(level: LoggerLevel, subTag: String, format: String, vararg arguments: Any)

    fun log(level: LoggerLevel, subTag: String, t: Throwable)

    fun json(level: LoggerLevel, msg: String)

    fun json(level: LoggerLevel, subTag: String, msg: String)
}